File: WPSoft.AVGCore/PluginInterface.vb
Imports WADV
Imports WADV.AppCore.Plugin

Namespace PluginInterface

    Public Class Script : Implements IScript

        Public Sub StartRegisting(ScriptVM As LuaInterface.Lua) Implements AppCore.Plugin.IScript.StartRegisting
            ScriptAPI.RegisterFunction(Reflection.Assembly.GetExecutingAssembly.GetTypes, "WPSoft.AVGCore.API", "AVG")
        End Sub

    End Class

End Namespace

File: WPSoft.ImageModule/PluginInterface.cpp
#include "Stdafx.h"
#include "PluginInterface.h"
#include "Config.h"

namespace PluginInterface {

	bool Initialise::StartInitialising() {
		Config::ImageConfig::ReadConfig();
		return true;
	}

	void Script::StartRegisting(LuaInterface::Lua^ ScriptVM) {
		ScriptAPI::RegisterFunction(System::Reflection::Assembly::GetExecutingAssembly()->GetTypes(), "API", "SIM");
	}

	bool Loop::StartLooping() {
		return true;
	}
	
	void Loop::StartRendering(System::Windows::Window^ window) {

	}

}